### PR TITLE
feat(php): use strings for request enums + other small fixes

### PR DESCRIPTION
**Pull request by dcb6**
changelog:
```yml
    - type: feat
      summary: >-
        Represent enums in objects as strings.
    - type: fix
      summary: >-
        Generated wrapped requests now implement `SerializableType`.
        Represent enums in objects as strings.
    - type: fix
      summary: >-
        Fix a bug where we don't set the request options baseurl properly.
    - type: fix
      summary: >-
        Fix bugs in our numeric type serde and add tests.
```

also:
* some initial support for unions, which is leftover from a previous enum handling implementation

### CODE REVIEW

File: generators/php/sdk/versions.yml [hunk 1]
```diff
@@ -1,11 +1,26 @@
+- version: 0.1.2
+  changelogEntry:
+    - type: feat
+      summary: >-
+        Represent enums in objects as strings.
+    - type: fix
+      summary: >-
+        Generated wrapped requests now implement `SerializableType`.
+        Represent enums in objects as strings.
```
Comment (amckinney) [inline generators/php/sdk/versions.yml:9]: Let's remove this (I know you meant to based on the entry above).
```suggestion
        Generated wrapped requests now implement `SerializableType`.
```

File: generators/php/codegen/src/project/PhpProject.ts [hunk 2]
```diff
@@ -238,7 +238,7 @@ class ComposerJson {
         return `
 {
   "name": "${composerProjectName}",
-  "version": ${this.context.version != null ? `"${this.context.version}"` : "0.0.0"},
+  "version": ${this.context.version != null ? `"${this.context.version}"` : '"0.0.0"'},
```
Comment (amckinney) [inline generators/php/codegen/src/project/PhpProject.ts:241]: Nice 👏

File: generators/php/codegen/src/context/PhpAttributeMapper.ts [hunk 3]
```diff
@@ -103,6 +111,8 @@ export class PhpAttributeMapper {
                     writer.write("::class");
                 });
             }
+            case "enumString":
+                return php.codeblock('"string"');
```
Comment (amckinney) [inline generators/php/codegen/src/context/PhpAttributeMapper.ts:115]: Should we prefer single quotes here?
```suggestion
                return php.codeblock("'string'");
```

File: generators/php/codegen/src/ast/Enum.ts [hunk 4]
```diff
@@ -48,8 +56,16 @@ export class Enum extends AstNode {
         if (this.backing != null) {
             writer.write(` : ${this.backing}`);
         }
+        if (this.serializable) {
+            writer.addReference(
+                new ClassReference({
+                    name: "JsonSerializable",
+                    namespace: ""
+                })
```
Comment (amckinney) [inline generators/php/codegen/src/ast/Enum.ts:64]: nit: We should make this a function on the abstract context, so we can just read:
```
writer.addReference(this.context.getJsonSerializableClassReference())
```

Check out [getJsonExceptionClassReference](https://github.com/fern-api/fern/blob/6d3572ac09858e9883b96fca458463347083a4bc/generators/php/sdk/src/SdkGeneratorContext.ts#L116) for an example.

File: seed/php-sdk/alias-extends/src/SeedClient.php [hunk 5]
```diff
@@ -67,7 +67,7 @@ public function extendedInlineRequestBody(InlinedChildRequest $request, ?array $
         try {
             $response = $this->client->sendRequest(
                 new JsonApiRequest(
-                    baseUrl: $this->options['baseUrl'] ?? $this->client->options['baseUrl'] ?? '',
```
Comment (amckinney) [inline seed/php-sdk/alias-extends/src/SeedClient.php:70]: ty 👏

File: generators/php/codegen/src/context/PhpTypeMapper.ts [hunk 6]
```diff
@@ -17,6 +17,11 @@ import { AbstractPhpGeneratorContext } from "./AbstractPhpGeneratorContext";
 export declare namespace PhpTypeMapper {
     interface Args {
         reference: TypeReference;
+        /*
+         * By default, we represent enums as strings, with a phpstan phpdoc referencing the generated enum. If this flag is
+         * is true, then we reference the enum type directly.
+         */
+        preserveEnums?: boolean;
```
Comment (amckinney) [inline generators/php/codegen/src/context/PhpTypeMapper.ts:24]: Is this still used? I don't see anywhere that sets it to `true`.

**Review by amckinney — APPROVED** [review]
Nice, only a few nits + suggestions!

File: generators/php/sdk/versions.yml (hunk 1 above)
Comment (dcb6) [reply generators/php/sdk/versions.yml:9]: oop thanks!

File: generators/php/codegen/src/ast/Enum.ts (hunk 4 above)
Comment (dcb6) [reply generators/php/codegen/src/ast/Enum.ts:64]: we don't have `context` in Ast classes though right?

File: generators/php/codegen/src/context/PhpTypeMapper.ts (hunk 6 above)
Comment (dcb6) [reply generators/php/codegen/src/context/PhpTypeMapper.ts:24]: nope not anymore! ok if we leave it around in case, or you want it gone?

**Pull request merged**